core/aws-core/src/main/java/software/amazon/awssdk/awscore/internal/AwsExecutionContextBuilder.java

```diff
@@ -124,13 +126,40 @@ private AwsExecutionContextBuilder() {
-            authorizationStrategy.addCredentialsToExecutionAttributes(executionAttributes);
-            signer = authorizationStrategy.resolveSigner();
+
+        // Note, not setting IS_NONE_AUTH_TYPE_REQUEST in newly generated clients, makes earlier logic not sufficient, as
+        // NoneAuthTypeRequestTest fails.
+        // It is easiest to keep NoneAuthTypeRequest passing by continuing to set IS_NONE_AUTH_TYPE_REQUEST in newly generated
+        // clients. This might become odd in the future, when authType=none is deprecated (authType itself will be less favored
+        // in lieu of authList, to support multiple auth schemes). But we can cross that bridge as part of those changes?
+
+        SelectedAuthScheme<?> selectedAuthScheme =
+            executionAttributes.getAttribute(SdkInternalExecutionAttribute.SELECTED_AUTH_SCHEME);
+        if (selectedAuthScheme == null) {
+            // this implies old (pre SRA) client, keeping the logic as before intact for that case.
+            if (isAuthenticatedRequest(executionAttributes)) {
+                AuthorizationStrategyFactory authorizationStrategyFactory =
+                    new AuthorizationStrategyFactory(interceptorContext.request(), metricCollector, clientConfig);
+                AuthorizationStrategy authorizationStrategy =
+                    authorizationStrategyFactory.strategyFor(executionParams.credentialType());
+                authorizationStrategy.addCredentialsToExecutionAttributes(executionAttributes);
+                signer = authorizationStrategy.resolveSigner();
+            }
+        } else {
+            // With SRA, i.e., for newly generated clients:
+            // If old Signer is still provided via overrides (existing customer code), we want to continue using that signer.
+            // However, we don't want to use that overridden signer if the operation was going to have no auth (authType=none
+            // in the model or auth scheme resolver selects the smithy.api#noAuth scheme (abstracted as
+            // selectedAuthScheme.supportsSigning), in which case we want Signer to end up as `null`.
+            if (selectedAuthScheme.supportsSigning() && SignerOverrideUtils.isSignerOverridden(interceptorContext.request(),
+                                                                                               executionAttributes)) {
+                AuthorizationStrategyFactory authorizationStrategyFactory =
+                    new AuthorizationStrategyFactory(interceptorContext.request(), metricCollector, clientConfig);
+                AuthorizationStrategy authorizationStrategy =
+                    authorizationStrategyFactory.strategyFor(executionParams.credentialType());
+                authorizationStrategy.addCredentialsToExecutionAttributes(executionAttributes);
+                signer = authorizationStrategy.resolveSigner();
+            }
```

Comment:
Can we replace this with 

~~~
        if (usesOverriddenSigner(selectedAuthScheme, interceptorContext.request(), executionAttributes) ||
            authenticatedWithoutAuthSchemeSupport(selectedAuthScheme)) {
            AuthorizationStrategyFactory authorizationStrategyFactory =
                new AuthorizationStrategyFactory(interceptorContext.request(), metricCollector, clientConfig);
            AuthorizationStrategy authorizationStrategy =
                authorizationStrategyFactory.strategyFor(executionParams.credentialType());
            authorizationStrategy.addCredentialsToExecutionAttributes(executionAttributes);
            signer = authorizationStrategy.resolveSigner();
        }
~~~
?
Coalescing the different conditions into boolean methods with names that remove the need for long comments. 
Maybe changing the order of the conditions wasn't great, it could be better to do the null auth scheme first